dask/typing.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from typing import Any, Callable, Mapping, MutableMapping, Protocol, runtime_checkable
+
+try:
+    from IPython.display import DisplayObject
+except ImportError:
+    DisplayObject = object
+
+
+@runtime_checkable
+class DaskCollection(Protocol):
+    def __dask_graph__(self) -> Mapping | None:
+        pass
+
+    def __dask_keys__(self) -> list:
+        pass
+
+    def __dask_postcompute__(self) -> tuple[Callable, tuple]:
+        pass
+
+    def __dask_postpersist__(self) -> tuple[Callable, tuple]:
+        pass
+
+    def __dask_tokenize__(self) -> Any:
+        pass
+
+    @staticmethod
+    def __dask_optimize__(dsk: MutableMapping, keys: Any, **kwargs) -> MutableMapping:
+        pass
+
+    @staticmethod
+    def __dask_scheduler__(dsk: Mapping, keys: Any, **kwargs) -> Any:
+        pass
+
+    def compute(self, **kwargs) -> Any:
+        pass
+
+    def persist(self, **kwargs) -> Any:
```

Comment:
This would be a great place to use a [`Self` type](https://www.python.org/dev/peps/pep-0673/), though it's only just been accepted and not widely available yet. Though there *is* a [workaround](https://www.python.org/dev/peps/pep-0673/#use-in-protocols) that works today using generics:

```python

CollectionType = TypeVar("CollectionType", bound=DaskCollection)

...


    def persist(self: CollectionType, **kwargs) -> CollectionType:
        pass
```